### ISSUE TITLE
Invalid CLI flag prints the usage text as a list of raw byte values

### ISSUE THREAD

**Issue by DonIsaac** [DonIsaac/zlint#379]
# Invalid CLI flag prints the usage text as a list of raw byte values

- **Severity:** Low-medium (cosmetic, but every user who typos a flag sees it)
- **Status:** Verified by reproduction
- **Location:** `src/main.zig:35-39`

## The bug

```zig
// main.zig:35
var opts = Options.parseArgv(stack_alloc, io, init.minimal.args, &err) catch {
    std.debug.print("{s}\n{any}\n", .{ err.message.borrow(), Options.usage });
    err.deinit(stack_alloc);
    return 1;
};
```

`Options.usage` is a `*const [31:0]u8`; the `{any}` specifier formats it as a slice of numbers rather than a string.

Verified output:

```
$ zlint --bogus-flag
unknown option: --bogus-flag

{ 85, 115, 97, 103, 101, 58, 32, 122, 108, 105, 110, 116, 32, 91, 111, 112, 116, 105, 111, 110, 115, 93, 32, 91, 60, 100, 105, 114, 115, 62, 93 }
```

(Those bytes spell `Usage: zlint [options] [<dirs>]`.)

A related latent issue in the same block: if `parseArgv` fails with `error.OutOfMemory` (from `opts.args.append`), `err` is never written, so `err.message.borrow()` and `err.deinit()` operate on undefined memory — the same class as report 02.

## User impact

The first thing a new user sees after mistyping a flag is a byte-array dump, which looks like a crash and hides the usage string entirely.

## Proposed fix

Use `{s}` for the usage string:

```zig
std.debug.print("{s}\n{s}\n", .{ err.message.borrow(), Options.usage });
```

For the OOM path, either initialize `err` to a valid default before calling `parseArgv`, or handle `error.OutOfMemory` separately (as `lintFile` does).
